src/main/java/by/itech/lab/supplier/service/impl/WarehouseServiceImpl.java

```diff
@@ -39,4 +63,94 @@ public WarehouseDto save(WarehouseDto dto) {
     public void delete(Long id) {
 
     }
+
+    @Override
+    @Transactional
+    public void acceptItems(final ApplicationDto appDto, final Long warehouseId) {
+        lock.lock();
+        try {
+            applicationService.changeStatus(appDto.getId(), ApplicationStatus.STARTED_PROCESSING);
+            final ApplicationDto appFromDb = applicationService.findById(appDto.getId());
```

Comment:
May we execute this line first, then use appFromDb.getId() across all method instead of appDto.getId()?